front/src/components/UserVacationsComponent/ShowUserVacations.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useEffect, useState} from 'react';
+import {UserVacationsResponse, UserScheduleRequestApi} from "@noadudai/scheduler-backend-client/api";
+import axios from "axios";
+import {useAuth0} from "@auth0/auth0-react";
+import {useMutation} from "@tanstack/react-query";
+import { IoTrashOutline } from "react-icons/io5";
+import { HiOutlinePencil } from "react-icons/hi";
+import Pagination from "../Pagination.tsx";
+import {GetUserFutureVacationsPaginationModel} from "@noadudai/scheduler-backend-client";
+import GetPagesUserVacations from "./GetPagesUserVacations.tsx";
+
+
+const ShowUserVacations = () => {
+    const {getAccessTokenSilently} = useAuth0();
+
+    const [itemsPerPage] = useState(8);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [totalNumberOfVacations, setTotalNumberOfVacations] = useState(1);
+
+    const ax = axios.create({
+        baseURL: `${import.meta.env.VITE_BACKEND_BASE_URL}`,
+    });
+
+    const userSchedulePreferenceRequestApi: UserScheduleRequestApi = new UserScheduleRequestApi(undefined, undefined, ax);
+
+    const changePage = (pageNumber: number) => {
+        setCurrentPage(pageNumber)
+    };
+
+    const userSchedulePrefReqPostRequestNumberVacations = useMutation({
+        mutationFn: async () =>
+        {
+            const token = await getAccessTokenSilently();
+
+            const response = await userSchedulePreferenceRequestApi.userSchedulePreferencesRequestGetNumberOfFutureVacationsPost({
+                headers: {
+                    Authorization: `Bearer ${token}`,
+                },
+            });
+
+            return response.data;
+        },
+        onSuccess: (numVacations) => {
+            setTotalNumberOfVacations(numVacations);
```

Comment:
this is called `State Duplication`.  `userSchedulePrefReqPostRequestNumberVacations` will cause a re-render